app/main.py

```diff
@@ -1,34 +1,210 @@
+                whole_field[(i, index)] = symbol
+
+        for location in whole_field:
+            ship = self.field.get(location)
+            if ship:
+                symbol = "*"
+                if ship.is_drowned:
+                    symbol = "x"
+                for deck in ship.decks:
+                    if not deck.is_alive:
+                        whole_field[(deck.row, deck.column)] = symbol
+
+        for index, location in enumerate(whole_field):
+            if index % 10 == 0:
+                print("")
+            print(f"{whole_field.get(location)}      ", end="")
+
+    def _validate_field(self) -> bool:
+        single_deck_ship = 0
+        double_deck_ship = 0
+        three_deck_ship = 0
+        four_deck_ship = 0
+
+        ships = set([ship for ship in self.field.values()])
+        for ship in ships:
+            ship_length = len(ship.decks)
+            if ship_length == 1:
+                single_deck_ship += 1
+                continue
+            if ship_length == 2:
+                double_deck_ship += 1
+                continue
+            if ship_length == 3:
+                three_deck_ship += 1
+                continue
+            if ship_length == 4:
+                four_deck_ship += 1
+
+        if (
+                single_deck_ship == 4 and double_deck_ship == 3
+                and three_deck_ship == 2 and four_deck_ship == 1
+        ):
+            if self.check_neighbours_cells(ships):
+                return True
+        return False
+
+    def check_neighbours_cells(self, ships: set[Ship]) -> bool:
+        for ship in ships:
+            locations = []
+            for deck in ship.decks:
+                locations.append((deck.row, deck.column))
+
+            for location in locations:
+                if not self.check_cell(locations, location, 0, -1):
+                    return False
+                if not self.check_cell(locations, location, 1, -1):
+                    return False
+                if not self.check_cell(locations, location, 1, 0):
+                    return False
+                if not self.check_cell(locations, location, 1, 1):
+                    return False
+                if not self.check_cell(locations, location, 0, 1):
+                    return False
+                if not self.check_cell(locations, location, -1, 1):
+                    return False
+                if not self.check_cell(locations, location, -1, 0):
+                    return False
+                if not self.check_cell(locations, location, -1, -1):
+                    return False
+        return True
+
+    def check_cell(
+            self,
+            locations: list[tuple],
+            location: tuple,
+            row: int,
+            column: int
+    ) -> bool:
+        if (location[0] + row, location[1] + column) in self.field:
+            if (location[0] + row, location[1] + column) not in locations:
+                return False
+        return True
```

Comment:
Rewrite this method with the use of `all`, one return statement, and without `if` conditions.